internal/cmd/iam/command_pool.go

```diff
@@ -49,3 +50,15 @@ func (c *identityPoolCommand) validArgs(cmd *cobra.Command, args []string) []str
 	provider, _ := cmd.Flags().GetString("provider")
 	return pcmd.AutocompleteIdentityPools(c.V2Client, provider)
 }
+
+func (c *identityPoolCommand) poolAndProviderNamesToIds(pool string, provider string) (string, string, error) {
```

Comment:
Why are these being grouped in the same function? What if someone passes a pool ID and provider name?

Reply:
When we call the Get API for resources like `iam pool` that take in a flag of another resource that can be a name or ID like `iam provider` and the API call fails, we don't know if the error because the `pool` was a name or invalid or if the `provider` flag value was a name or invalid. Therefore we would have to try to convert all of the potential names associated with an execution, starting with the flag(s) since the resource API call depends on the flags. I handle this here: https://github.com/confluentinc/cli/pull/2091/commits/00909f10938acb7cdef362feeb200cbb038aafb7#diff-b793b1ec65c5190a9e06fdb8284e914a0241bb31018ffbc9b81dea7f248cd090R36-R56, and I think first trying a GET request with the resource is the best thing to do here since we don't know if it's valid or not, but if it is, we avoid the list API call.